android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/MeasureReportViewModel.kt

```diff
@@ -360,164 +327,188 @@ constructor(
-                        Coding(
-                          GroupType.MEDICATION.system,
-                          GroupType.MEDICATION.toCode(),
-                          GroupType.MEDICATION.display
-                        )
-                      )
-                  }
-                )
-              }
-              .map {
-                fhirOperator.evaluateMeasure(
-                  measureUrl = measureUrl,
-                  start = startDateFormatted,
-                  end = endDateFormatted,
-                  reportType = POPULATION,
-                  subject = it.id,
-                  practitioner = null
-                  /* TODO DO NOT pass this id to MeasureProcessor as this is treated as subject if subject is null.
-                  practitionerId?.asReference(ResourceType.Practitioner)?.reference*/ ,
-                  lastReceivedOn = null // Non-null value not supported yet
-                )
-              }
-              .first()
-          } else
-            fhirOperator.evaluateMeasure(
-              measureUrl = measureUrl,
-              start = startDateFormatted,
-              end = endDateFormatted,
-              reportType = POPULATION,
-              subject = null,
-              practitioner = null
-              /* TODO DO NOT pass this id to MeasureProcessor as this is treated as subject if subject is null.
-              practitionerId?.asReference(ResourceType.Practitioner)?.reference*/ ,
-              lastReceivedOn = null // Non-null value not supported yet
-            )
-        } catch (exception: IllegalArgumentException) {
-          Timber.e(exception)
-          null
+    subjectXFhirQuery: String?
+  ): List<MeasureReport> {
+    val measureReport = mutableListOf<MeasureReport>()
+    withContext(dispatcherProvider.io()) {
+      if (subjectXFhirQuery?.isNotEmpty() == true) {
+        fhirEngine
+          .search(subjectXFhirQuery)
+          .map {
+            // TODO a hack to prevent missing subject in case of Group based reports where
+            // MeasureEvaluator looks for Group members and skips the Group itself
+            if (it is Group && !it.hasMember()) {
+              it.addMember(Group.GroupMemberComponent(it.asReference()))
+              fhirEngine.update(it)
+            }
+            runMeasureReport(measureUrl, SUBJECT, startDateFormatted, endDateFormatted, it.id)
+          }
+          .forEach { measureReport.add(it) }
+      } else
+        runMeasureReport(measureUrl, POPULATION, startDateFormatted, endDateFormatted, null).also {
+          measureReport.add(it)
         }
-      }
 
-    if (measureReport != null) {
-      defaultRepository.addOrUpdate(resource = measureReport)
-      _measureReportPopulationResultList.addAll(
-        formatPopulationMeasureReport(measureReport, indicatorTitle)
-      )
+      measureReport.forEach { defaultRepository.addOrUpdate(resource = it) }
     }
+    return measureReport
+  }
+
+  fun runMeasureReport(
```

Comment:
could you please add a docstring

Reply:
it is here. I do not know why it does not show it as outdated into main view